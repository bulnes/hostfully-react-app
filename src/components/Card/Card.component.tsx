import { HouseProps } from "../../@types/HouseProps";

interface CardProps extends HouseProps {}

export function Card({
  id,
  title,
  description,
  pricePerNight,
  maxGuests,
  bedrooms,
  bathrooms,
  image,
}: CardProps) {
  return (
    <div className="bg-white border rounded-lg shadow-md">
      <div className="flex justify-center pb-5">
        <img
          className="rounded-t-lg"
          src={image}
          alt=""
          loading="lazy"
          width={478}
          height={319}
        />
      </div>

      <div className="px-4 pb-3">
        <h5 className="text-xl font-semibold tracking-tight hover:text-violet-800 text-gray-900 mb-3">
          {title}
        </h5>

        <p className="text-gray-600 text-sm break-all mb-4">{description}</p>

        <div>
          <div className="flex items-start lg:items-center justify-between flex-col lg:flex-row">
            <p className="text-gray-900 font-semibold">
              ${pricePerNight} <span className="text-gray-600">/ night</span>
            </p>

            <p className="flex items-center gap-3">
              <span className="text-gray-600">{maxGuests} guests</span>
              <span className="text-gray-600"> • </span>
              <span className="text-gray-600">{bedrooms} bedrooms</span>
              <span className="text-gray-600"> • </span>
              <span className="text-gray-600">{bathrooms} baths</span>
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}
