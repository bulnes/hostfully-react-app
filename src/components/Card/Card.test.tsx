import { render } from "@testing-library/react";
import { Card } from "./Card.component";

describe("Component: Card", () => {
  const mockHouseProps = {
    id: 0,
    title: "Sample Title",
    description: "Sample Description",
    pricePerNight: 100,
    maxGuests: 4,
    bedrooms: 2,
    bathrooms: 2,
    image: "sample-image.jpg",
  };

  it("should be defined", () => {
    expect(Card).toBeDefined();
  });

  it("should render correctly", () => {
    const wrapper = render(<Card {...mockHouseProps} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
