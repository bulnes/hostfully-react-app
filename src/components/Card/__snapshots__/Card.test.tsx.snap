// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Card should be represented by the snapshot 1`] = `
<div>
  <div
    class="bg-white border rounded-lg shadow-md"
  >
    <div
      class="flex justify-center pb-5"
    >
      <img
        alt=""
        class="rounded-t-lg"
        height="319"
        loading="lazy"
        src="https://via.placeholder.com/478x319"
        width="478"
      />
    </div>
    <div
      class="px-4 pb-3"
    >
      <h5
        class="text-xl font-semibold tracking-tight hover:text-violet-800 text-gray-900 mb-3"
      >
        Title
      </h5>
      <p
        class="text-gray-600 text-sm break-all mb-4"
      >
        Description
      </p>
      <div>
        <div
          class="flex items-start lg:items-center justify-between flex-col lg:flex-row"
        >
          <p
            class="text-gray-900 font-semibold"
          >
            $
            100
             
            <span
              class="text-gray-600"
            >
              / night
            </span>
          </p>
          <p
            class="flex items-center gap-3"
          >
            <span
              class="text-gray-600"
            >
              4
               guests
            </span>
            <span
              class="text-gray-600"
            >
               • 
            </span>
            <span
              class="text-gray-600"
            >
              2
               bedrooms
            </span>
            <span
              class="text-gray-600"
            >
               • 
            </span>
            <span
              class="text-gray-600"
            >
              2
               baths
            </span>
          </p>
        </div>
        <hr
          class="my-4"
        />
        <form>
          <div
            class="flex items-center gap-3 mt-3 mb-5"
          >
            <div
              class="flex items-start flex-col gap-1"
            >
              <label
                class="text-gray-600"
                for="checkin"
              >
                Check-in
              </label>
              <input
                class="border rounded-md p-1"
                id="checkin"
                type="date"
                value=""
              />
            </div>
            <div
              class="flex items-start flex-col gap-1"
            >
              <label
                class="text-gray-600"
                for="checkout"
              >
                Check-out
              </label>
              <input
                class="border rounded-md p-1 w-full"
                id="checkout"
                type="date"
                value=""
              />
            </div>
          </div>
          <button
            class="bg-blue-500 hover:bg-blue-700 text-white border border-blue-500 font-bold py-2 px-4 rounded"
            type="submit"
          >
            Book now
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
